### PR TITLE
DOC: test organization

### PR DESCRIPTION
**Pull request by jbrockmendel**
Documenting the MO behind the test organization we've been doing.

1) I expect this is not valid rst ATM
2) The style does not match the rest of the docs, so probably needs a re-write.

### CODE REVIEW

File: doc/source/development/test_writing.rst [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        - tests.plotting
+
+    B) Is the method an IO method?
+    This test likely belongs in one of:
+
+        - tests.io
+
+    C) Is the method an indexing method?
+        i) Is the method ``loc``, ``iloc``, ``at``, or ``iat``?
+        This test likely belongs in one of:
```
Comment (jreback) [inline doc/source/development/test_writing.rst:64]: can you be a little more precise here (e.g. do we have criteria for putting in pandas/tests/indexing over the frame/series ones? (can be a TODO of course)

File: doc/source/development/index.rst [hunk 2]
```diff
@@ -21,3 +21,4 @@ Development
     policies
     roadmap
     meeting
+    test_writing
```
Comment (jreback) [inline doc/source/development/index.rst:24]: maybe put after internals

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm small comments.

**Comment by jreback** [comment]
thanks @jbrockmendel very nice

**Comment by jbrockmendel** [comment]
i think this might have been merged prematurely, the docbuild is complaining about rst stuff i need to figure out

**Comment by jorisvandenbossche** [comment]
@jbrockmendel if you push your branch again, I *think* you should be able to reopen the PR (unless you also already deleted the branch locally), and then we can work out the failures here

**Pull request merged**